src/gluonts/torch/model/predictor.py

```diff
@@ -73,9 +74,11 @@ def network(self) -> nn.Module:
     def predict(
         self, dataset: Dataset, num_samples: Optional[int] = None
     ) -> Iterator[Forecast]:
-        self.input_transform += SelectFields(
-            self.input_names + self.required_fields, allow_missing=True
-        )
+        if self.provided_fields:
+            self.input_transform += SelectFields(
+                self.input_names + self.required_fields, allow_missing=True
+            )
+            self.provided_fields = True
         inference_data_loader = InferenceDataLoader(
             dataset,
             transform=self.input_transform,
```

Comment:
@Cameronwood611 the problem is with the `+=` which updates the predictor state every time `.predict` is invoked. Solution should be to either do here

```python
input_transform = self.input_transform + SelectFields(...)
inference_data_loader = InferenceDataLoader(
    ...,
    transform=input_transform,
    ....,
)
```

or directly in the constructor 

```
self.input_transform = input_transform + SelectFields(...)
```

and remove lines 76-78 from here.